be/src/exec/vectorized/chunks_sorter_topn.cpp

```diff
@@ -460,4 +464,28 @@ Status ChunksSorterTopn::_hybrid_sort_first_time(RuntimeState* state, Permutatio
     return Status::OK();
 }
 
+void ChunksSorterTopn::_rank_pruning() {
+    if (_topn_type != TTopNType::RANK) {
+        return;
+    }
+    if (_merged_segment.order_by_columns.empty()) {
+        return;
+    }
+    if (_merged_segment.chunk->num_rows() <= _get_number_of_rows_to_sort()) {
+        return;
+    }
+    const auto peer_group_start = static_cast<int>(_get_number_of_rows_to_sort()) - 1;
```

Comment:
Is the `int` suitable for this scenario ?

Reply:
In case of `_get_number_of_rows_to_sort()` returns 0, `0-1` in size_t will overflow